src/common/thread.pool.cpp

```diff
@@ -0,0 +1,155 @@
+#include "thread.pool.hh"
+#include "utilities.hh"
+
+namespace zarr = acquire::sink::zarr;
+namespace common = zarr::common;
+
+common::ThreadPool::ThreadPool(size_t n_threads,
+                               std::function<void(const std::string&)> err)
+  : error_handler_{ err }
+  , is_accepting_jobs_{ true }
+{
+    n_threads = std::clamp(
```

Comment:
1. Explicit `static_cast` is preferred in modern C++ because it clearly indicates the type of conversion.
2. Use `size_t` consistently instead of mixing `unsigned`.
```cpp
n_threads = std::clamp(n_threads, static_cast<std::size_t>(1),
            std::max(std::thread::hardware_concurrency(), static_cast<std::size_t>(1)));
```

Reply:
What we have happening here is `std::thread::hardware_concurrency()` returning `unsigned int` (usually uint32), which isn't a `size_t` in general. The only reason we're using `size_t` here at all is because that's what the constructor asks for. It would actually be better to modify it constructor to take `unsigned int`, so we don't have to do so much casting.

Reply:
You can use unsigned literals (i.e., `1u`) instead of creating a trivial variable.